src/products/controller/stocks.controller.ts

```diff
@@ -0,0 +1,63 @@
+import { Body, Controller, Get, Param, Post, Put } from '@nestjs/common';
+import { StocksService } from '../service/stocks.service';
+import { StockMapper } from '../mapper/stock.mapper';
+import { CreateStockDTO } from '../dto/create-stock.dto';
+import { StockDTO } from '../dto/stock.dto';
+import { Stock } from '../domain/stock.domain';
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
+
+@ApiTags('stocks')
+@Controller('stocks')
+export class StocksController {
+  constructor(
+    private readonly stocksService: StocksService,
+    private stockMapper: StockMapper,
+  ) {}
+
+  @Post()
+  @ApiResponse({
+    status: 201,
+    description: 'The stock has been successfully created.',
+  })
+  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  async create(@Body() createStockDto: CreateStockDTO): Promise<Stock> {
+    const stock = this.stockMapper.toDomain(createStockDto);
+    return this.stocksService.create(stock);
+  }
+
+  @Get()
+  @ApiResponse({
+    status: 200,
+    description: 'All stocks retrieved successfully.',
+  })
+  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  async findAll(): Promise<StockDTO[]> {
+    const stocks = await this.stocksService.findAll();
+    return stocks.map((stock) => this.stockMapper.toDTO(stock));
+  }
+
+  @Get(':locationId/:productId')
```

Comment:
Nice path binding for these IDs!

You did some extra work here. This controller for instance was not required. The stock should have been updated via the order processing flow.